ui/core/proto_utils/equipped_item.ts

```diff
@@ -219,23 +300,50 @@ export class EquippedItem {
 	}
 
 	withRandomSuffix(randomSuffix: ItemRandomSuffix | null): EquippedItem {
-		return new EquippedItem(this._item, this._enchant, this._gems, randomSuffix, this._reforge);
+		return new EquippedItem({
+			item: this._item,
+			enchant: this._enchant,
+			gems: this._gems,
+			randomSuffix,
+			reforge: this._reforge,
+			upgrade: this._upgrade,
+		});
 	}
 
 	getWithRandomSuffixStats() {
+		return this._getWithDynamicStats();
+	}
+
+	getRandomPropPoints(): number {
+		return this.item.scalingOptions[this._upgrade]?.randPropPoints || this.item.randPropPoints;
+	}
+
+	getWithUpgradeStats() {
+		return this._getWithDynamicStats();
+	}
+
+	_getWithDynamicStats() {
```

Comment:
Since `getWithRandomSuffixStats()` and `getWithUpgradeStats()` are both identical aliases to `_getWithDynamicStats()`, it would be cleaner to just make `getWithDynamicStats()` public and delete the two aliases.